internal/pkg/service/stream/storage/level/local/encoding/manager.go

```diff
@@ -0,0 +1,158 @@
+package encoding
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	"github.com/benbjohnson/clock"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/servicectx"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/events"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type Manager struct {
```

Comment:
`encoding.Manager`, very similar to `diskwriter.Volume` and `diskreader.Volume` code.